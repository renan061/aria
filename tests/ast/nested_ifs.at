--------------------------------------------------------------------------------
-- Title
Nested If's and If-Else's
-- Input
function main {
    // ==================================================
    //
    //  Empty blocks
    //
    // ==================================================

    // if-else
    if true {} else {}

    // if-elseif
    if true {} else if true {}

    // if-elseif-else
    if true {} else if true {} else {}

    // if-elseif-elseif-elseif
    if true {} else if true {} else if true {} else if true {}

    // if-elseif-elseif-elseif-else
    if true {} else if true {} else if true {} else if true {} else {}

    // ==================================================
    //
    //  Blocks with content
    //
    // ==================================================

    // if-else
    if true {
        a := 1; b := 2; c := 3;
    } else {
        a := 1; b := 2; c := 3;
    }

    // if-elseif
    if true {
        a := 1; b := 2; c := 3;
    } else if true {
        a := 1; b := 2; c := 3;
    }

    // if-elseif-else
    if true {
        a := 1; b := 2; c := 3;
    } else if true {
        a := 1; b := 2; c := 3;
    } else {
        a := 1; b := 2; c := 3;
    }

    // if-elseif-elseif-elseif-else
    if true {
        a := 1; b := 2; c := 3;
    } else if true {
        a := 1; b := 2; c := 3;
    } else if true {
        a := 1; b := 2; c := 3;
    } else if true {
        a := 1; b := 2; c := 3;
    }

    // if-elseif-elseif-elseif-else
    if true {
        a := 1; b := 2; c := 3;
    } else if true {
        a := 1; b := 2; c := 3;
    } else if true {
        a := 1; b := 2; c := 3;
    } else if true {
        a := 1; b := 2; c := 3;
    } else {
        a := 1; b := 2; c := 3;
    }

    // ==================================================
    //
    //  Blocks with if/elseif/else
    //
    // ==================================================

    if true {
        if true {}
    }

    if true {
        if true {} else {}
    }

    if true {} else if true {
        if true {} else {}
    } else {
        if true {} else if true {}
    }

    // if-elseif-elseif-elseif-else
    if true {
    } else {
        if true {
        } else {
            if true {
            } else {
                if true {
                } else {
                }
            }
        }
    }
}
-- Output
function main: Void {
    if (true) {
    } else {
    }
    if (true) {
    } else {
        if (true) {
        }
    }
    if (true) {
    } else {
        if (true) {
        } else {
        }
    }
    if (true) {
    } else {
        if (true) {
        } else {
            if (true) {
            } else {
                if (true) {
                }
            }
        }
    }
    if (true) {
    } else {
        if (true) {
        } else {
            if (true) {
            } else {
                if (true) {
                } else {
                }
            }
        }
    }
    if (true) {
        value a: ? = (1)
        value b: ? = (2)
        value c: ? = (3)
    } else {
        value a: ? = (1)
        value b: ? = (2)
        value c: ? = (3)
    }
    if (true) {
        value a: ? = (1)
        value b: ? = (2)
        value c: ? = (3)
    } else {
        if (true) {
            value a: ? = (1)
            value b: ? = (2)
            value c: ? = (3)
        }
    }
    if (true) {
        value a: ? = (1)
        value b: ? = (2)
        value c: ? = (3)
    } else {
        if (true) {
            value a: ? = (1)
            value b: ? = (2)
            value c: ? = (3)
        } else {
            value a: ? = (1)
            value b: ? = (2)
            value c: ? = (3)
        }
    }
    if (true) {
        value a: ? = (1)
        value b: ? = (2)
        value c: ? = (3)
    } else {
        if (true) {
            value a: ? = (1)
            value b: ? = (2)
            value c: ? = (3)
        } else {
            if (true) {
                value a: ? = (1)
                value b: ? = (2)
                value c: ? = (3)
            } else {
                if (true) {
                    value a: ? = (1)
                    value b: ? = (2)
                    value c: ? = (3)
                }
            }
        }
    }
    if (true) {
        value a: ? = (1)
        value b: ? = (2)
        value c: ? = (3)
    } else {
        if (true) {
            value a: ? = (1)
            value b: ? = (2)
            value c: ? = (3)
        } else {
            if (true) {
                value a: ? = (1)
                value b: ? = (2)
                value c: ? = (3)
            } else {
                if (true) {
                    value a: ? = (1)
                    value b: ? = (2)
                    value c: ? = (3)
                } else {
                    value a: ? = (1)
                    value b: ? = (2)
                    value c: ? = (3)
                }
            }
        }
    }
    if (true) {
        if (true) {
        }
    }
    if (true) {
        if (true) {
        } else {
        }
    }
    if (true) {
    } else {
        if (true) {
            if (true) {
            } else {
            }
        } else {
            if (true) {
            } else {
                if (true) {
                }
            }
        }
    }
    if (true) {
    } else {
        if (true) {
        } else {
            if (true) {
            } else {
                if (true) {
                } else {
                }
            }
        }
    }
}