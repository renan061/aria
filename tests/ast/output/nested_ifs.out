{
	function foo {
		if (true) {
		} else {
		}
		if (true) {
		} else {
			if (true) {
			}
		}
		if (true) {
		} else {
			if (true) {
			} else {
			}
		}
		if (true) {
		} else {
			if (true) {
			} else {
				if (true) {
				} else {
					if (true) {
					}
				}
			}
		}
		if (true) {
		} else {
			if (true) {
			} else {
				if (true) {
				} else {
					if (true) {
					} else {
					}
				}
			}
		}
		if (true) {
			a: ? := (1)
			b: ? := (2)
			c: ? := (3)
		} else {
			a: ? := (1)
			b: ? := (2)
			c: ? := (3)
		}
		if (true) {
			a: ? := (1)
			b: ? := (2)
			c: ? := (3)
		} else {
			if (true) {
				a: ? := (1)
				b: ? := (2)
				c: ? := (3)
			}
		}
		if (true) {
			a: ? := (1)
			b: ? := (2)
			c: ? := (3)
		} else {
			if (true) {
				a: ? := (1)
				b: ? := (2)
				c: ? := (3)
			} else {
				a: ? := (1)
				b: ? := (2)
				c: ? := (3)
			}
		}
		if (true) {
			a: ? := (1)
			b: ? := (2)
			c: ? := (3)
		} else {
			if (true) {
				a: ? := (1)
				b: ? := (2)
				c: ? := (3)
			} else {
				if (true) {
					a: ? := (1)
					b: ? := (2)
					c: ? := (3)
				} else {
					if (true) {
						a: ? := (1)
						b: ? := (2)
						c: ? := (3)
					}
				}
			}
		}
		if (true) {
			a: ? := (1)
			b: ? := (2)
			c: ? := (3)
		} else {
			if (true) {
				a: ? := (1)
				b: ? := (2)
				c: ? := (3)
			} else {
				if (true) {
					a: ? := (1)
					b: ? := (2)
					c: ? := (3)
				} else {
					if (true) {
						a: ? := (1)
						b: ? := (2)
						c: ? := (3)
					} else {
						a: ? := (1)
						b: ? := (2)
						c: ? := (3)
					}
				}
			}
		}
		if (true) {
			if (true) {
			}
		}
		if (true) {
			if (true) {
			} else {
			}
		}
		if (true) {
		} else {
			if (true) {
				if (true) {
				} else {
				}
			} else {
				if (true) {
				} else {
					if (true) {
					}
				}
			}
		}
		if (true) {
		} else {
			if (true) {
			} else {
				if (true) {
				} else {
					if (true) {
					} else {
					}
				}
			}
		}
	}
}
